### PR TITLE
Issue #1531 - Pagination to be implemented in all widgets - For better user experience 

### PR DESCRIPTION
**Pull request by Varun270489**
implemented pagination in api-list & product-list widget

### CODE REVIEW

File: src/components/apis/list-of-apis/ko/runtime/api-list.html [hunk 1]
```diff
@@ -127,28 +129,68 @@
...
+            </li>
         </ul>
     </div>
     <!-- /ko -->
     <!-- /ko -->
-</div>
\ No newline at end of file
+</div>
+
+<style>
```
Comment (azaslonov) [inline src/components/apis/list-of-apis/ko/runtime/api-list.html:161]: Could you please move this <style> block content to [pagination.scss](https://github.com/Azure/api-management-developer-portal/blob/master/src/themes/website/styles/pagination.scss) file and see how existing styles can be married with yours?

File: src/components/apis/list-of-apis/ko/runtime/api-list.html [hunk 2]
```diff
@@ -127,28 +129,68 @@
     <!-- ko if: hasPager -->
```
Comment (azaslonov) [inline src/components/apis/list-of-apis/ko/runtime/api-list.html:129]: `<!-- ko if: hasPager -->` and `data-bind="visible: needPaging"` seem like solving the same problem. Probably having `<!-- ko if: needPaging -->` on top level would be enough.

File: src/components/apis/list-of-apis/ko/runtime/api-list.ts [hunk 3]
```diff
@@ -120,6 +124,140 @@ export class ApiList {
         }
     }
 
+     //-----pagination changes start ---------------------------- 
+     public pageCount() {
```
Comment (azaslonov) [inline src/components/apis/list-of-apis/ko/runtime/api-list.ts:128]: `public pageCount() {` - Please add return types to every function.

File: src/components/apis/list-of-apis/ko/runtime/api-list.ts [hunk 4]
```diff
@@ -120,6 +124,140 @@ export class ApiList {
         }
     }
 
+     //-----pagination changes start ----------------------------
```
Comment (azaslonov) [inline src/components/apis/list-of-apis/ko/runtime/api-list.ts:127]: `//-----pagination changes start ----------------------------` - Please remove.

File: src/components/apis/list-of-apis/ko/runtime/api-list.ts [hunk 5]
```diff
@@ -120,6 +124,140 @@ export class ApiList {
...
+        this.page(next);
+        this.loadPageOfApis();
+    };
+
+    public goToLast() {
+        this.page(this.lastPage());
+        this.loadPageOfApis();
+    };
+
+    //------pagination changes end ---------------------------
```
Comment (azaslonov) [inline src/components/apis/list-of-apis/ko/runtime/api-list.ts:259]: `//------pagination changes end ---------------------------` - Please remove this comment

File: src/components/products/product-list/ko/runtime/product-list.html [hunk 6]
```diff
@@ -48,29 +50,69 @@
...
+            </li>
         </ul>
     </div>
     <!-- /ko -->
     <!-- /ko -->
-</div>
\ No newline at end of file
+</div>
+
+<style>
```
Comment (azaslonov) [inline src/components/products/product-list/ko/runtime/product-list.html:83]: Same as previous comment - please move this `<style>` tag content to designated file.

File: src/components/products/product-list/ko/runtime/product-list.ts [hunk 7]
```diff
@@ -44,6 +45,7 @@ export class ProductList {
         this.hasPrevPage = ko.observable(false);
         this.hasNextPage = ko.observable(false);
         this.hasPager = ko.computed(() => this.hasPrevPage() || this.hasNextPage());
+        this.totalNoOfItems = ko.observable();  //pagination changes
```
Comment (azaslonov) [inline src/components/products/product-list/ko/runtime/product-list.ts:48]: Please remove the comment `//pagination changes`

File: src/components/products/product-list/ko/runtime/product-list.ts [hunk 8]
```diff
@@ -27,6 +27,7 @@ export class ProductList {
     public readonly hasPager: ko.Computed<boolean>;
     public readonly hasPrevPage: ko.Observable<boolean>;
     public readonly hasNextPage: ko.Observable<boolean>;
+    public totalNoOfItems: ko.Observable<number>;   //pagination changes
```
Comment (azaslonov) [inline src/components/products/product-list/ko/runtime/product-list.ts:30]: Please remove the comment `//pagination changes`

File: src/components/products/product-list/ko/runtime/product-list.ts [hunk 9]
```diff
@@ -97,6 +99,139 @@ export class ProductList {
         }
     }
 
+     //-----pagination changes start ----------------------------
```
Comment (azaslonov) [inline src/components/products/product-list/ko/runtime/product-list.ts:102]: Please remove the comment `//-----pagination changes start ----------------------------`

File: src/services/apiService.ts [hunk 10]
```diff
@@ -201,7 +201,7 @@ export class ApiService {
 
         page.value = Object.keys(tagGroups).map(x => tagGroups[x]);
         page.nextLink = pageOfApiTagResources.nextLink;
-
+        page.count =pageOfApiTagResources.count;
```
Comment (azaslonov) [inline src/services/apiService.ts:204]: `page.count =pageOfApiTagResources.count` - please use Shift+Alt+F for code formatting.

**Review by azaslonov — CHANGES_REQUESTED** [review]

**Comment by Varun270489** [comment]
all review comments implemented

**Comment by azaslonov** [comment]
@Varun270489, considering you created another pull request, do we still need this one?

**Comment by Varun270489** [comment]
this is the only pull request

**Review by azaslonov — APPROVED** [review]

**Pull request merged**